src/query/graphite/native/builtin_functions.go

```
@@ -94,6 +95,43 @@ func sortByMaxima(ctx *common.Context, series singlePathSpec) (ts.SeriesList, er
 	return highestMax(ctx, series, len(series.Values))
 }
 
+/*
+Compares the maximum of each series against the given `value`. If the series
+maximum is greater than `value`, the regular expression search and replace is
+applied against the series name to plot a related metric
+
+e.g. given useSeriesAbove(ganglia.metric1.reqs,10,'reqs','time'),
+the response time metric will be plotted only when the maximum value of the
+corresponding request/s metric is > 10
+
+&target=useSeriesAbove(ganglia.metric1.reqs,10,"reqs","time")
+*/
+func useSeriesAbove(ctx *common.Context, seriesList singlePathSpec , maxAllowedValue float64, search, replace string) (ts.SeriesList, error) {
+	results := make([]*ts.Series, 0, len(seriesList.Values))
+	for _, series := range seriesList.Values {
+		if series.SafeMax() > maxAllowedValue {
+			opts := storage.FetchOptions{
+				StartTime: ctx.StartTime,
+				EndTime:   ctx.EndTime,
+				DataOptions: storage.DataOptions{
+					Timeout: time.Minute,
+				},
+			}
+			seriesName := strings.Replace(series.Name(), search, replace, 1)
+			result, _ := ctx.Engine.FetchByQuery(ctx, seriesName, opts)
```

Should check the error returned as second arg, and if it returns an error, this should return an error too.

---

Of course. Fixed.